### PR TITLE
mpl2: map insts and bterms to their cluster through unorderedmaps instead of dbIntProperty

### LINKED ISSUE #4765: Improve multilevel autoclustering performance in mpl2
### Description

Profiling showed that the multilevel autoclustering engine can have its runtime improved, specially due to the dbIntProperty mechanism responsible for mapping insts and bterms to their cluster.

### Suggested Solution

Change the dbIntProperty mechanism for an actual map.

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by AcKoucher**
Resolve #4765

Also renaming some variables..

### CODE REVIEW

File: src/mpl2/src/hier_rtlmp.cpp [hunk 1]
```diff
@@ -1494,73 +1491,74 @@ void HierRTLMP::mergeClusters(std::vector<Cluster*>& candidate_clusters)
...
     if (net->getSigType().isSupply()) {
       continue;
     }
-    int driver_id = -1;         // cluster id of the driver instance
-    std::vector<int> loads_id;  // cluster id of sink instances
-    bool pad_flag = false;
-    // check the connected instances
+
+    int driver_id = -1;
+    std::vector<int> loads_id;
```
Comment (maliberty) [inline src/mpl2/src/hier_rtlmp.cpp:1506]: Without the comment these might be clearer:

driver_id -> driver_cluster_id 
loads_id -> loads_cluster_id

File: src/mpl2/src/hier_rtlmp.h [hunk 2]
```diff
@@ -426,20 +421,13 @@ class HierRTLMP
...
-  // we only consider bus when we do bus planning
-  const int bus_net_threshold_ = 32;
-  // the weight used for balance timing and congestion
-  float congestion_weight_ = 0.5;
-
-  // Determine if the cluster is macro dominated
-  // if num_std_cell * macro_dominated_cluster_threshold_ < num_macro
-  // then the cluster is macro-dominated cluster
+  int large_net_threshold_ = 100;     // ignore global nets when clustering
+  const int bus_net_threshold_ = 32;  // only or bus planning
```
Comment (QuantamHD) [inline src/mpl2/src/hier_rtlmp.h:428]: Typo or -> for

**Comment by AcKoucher** [comment]
Secure-CI passed. (There were some small failures, but they were on master as well and were already fixed.)

**Comment by AcKoucher** [comment]
@maliberty Running Secure-CI again with stable CI.

**Comment by AcKoucher** [comment]
Secure-CI passed.

**Review by maliberty — APPROVED** [review]

**Pull request merged**